app/controllers/application_controller.rb

```diff
@@ -67,6 +67,24 @@ def get_background_id(sample)
 
   private
 
+  # Modified from https://gist.github.com/josevalim/fb706b1e933ef01e4fb6
+  def authenticate_user_from_token!
+    user_email = request.headers['X-User-Email'] || params[:user_token]
```

Comment:
I don't think we need `.presence` at all because a missed hash key is evaluated as `nil` in ruby.

Reply:
I thought the idea was preventing strings containing only spaces or even empty strings from being tested in our database.
```
irb(main):306:0> "" && "empty space is not really empty"
=> "empty space is not really empty"
irb(main):307:0> nil && "nihilism says that existence or values are meaningless."
=> nil
irb(main):308:0> "".presence && "nothing actually exists"
=> nil
```
So `.presence` would actually prevent this line from being executed in that context:
```
user = user_email && User.find_by(email: user_email)
```
This can be a problem if we have a user record in the database with an empty email and an empty token (our database model actually accepts that), but I hope that is not the case.

Reply:
Interesting, that's good to remember about ruby! 

About the empty values in the database, do you think

```
validates :email, presence: true
validates :authentication_token, presence: true
```

in `user.rb` would be enough, or should we validate in mysql itself?

Reply:
I've already checked the database for possible empty values and there are zero rows. I think we don't have a security issue in this case (assuming nobody is directly inserting such line into the table), and it would only be more related to avoid an extra query.
Looks fine for now. And I agree with Tiago, we should revisit this entire authentication mechanism.